common/helpers/sign.c

```diff
@@ -72,5 +73,66 @@ char* iota_sign_signature_gen(char const* const seed, size_t const index,
   return (char*)signature;
 }
 
+IOTA_EXPORT flex_trit_t* iota_flex_sign_address_gen(
+    flex_trit_t const* const seed, size_t const index, size_t const security) {
+  if (!(security > 0 && security <= 3)) return NULL;
+  const size_t key_length = security * ISS_KEY_LENGTH;
+
+  Kerl kerl;
+
+  trit_t subseed[HASH_LENGTH];
+  trit_t key[key_length];
+  flex_trit_t* address =
+      calloc(flex_trits_num_for_trits(HASH_LENGTH) + 1, sizeof(flex_trit_t));
+
+  init_kerl(&kerl);
+
+  flex_trit_array_to_int8(subseed, HASH_LENGTH, seed, HASH_LENGTH, HASH_LENGTH);
+  iss_kerl_subseed(subseed, subseed, index, &kerl);
+
+  iss_kerl_key(subseed, key, key_length, &kerl);
+  memset(subseed, 0, HASH_LENGTH * sizeof(trit_t));
+
+  iss_kerl_key_digest(key, key, key_length, &kerl);
+  iss_kerl_address(key, key, security * HASH_LENGTH, &kerl);
+
+  kerl_reset(&kerl);
+
+  int8_to_flex_trit_array(address, HASH_LENGTH, key, key_length, HASH_LENGTH);
+  memset(key, 0, key_length * sizeof(trit_t));
```

Comment:
Is it really useful ?

Reply:
I remember reading somewhere that it's good practice, to avoid "guessing" the key from the "noise" generated by the function. It was there before, I've left it untouched.

Reply:
Interesting, thanks